uber/reports.py

```diff
@@ -0,0 +1,65 @@
+from uber.common import *
+
+
+class ReportBase:
+    def __init__(self, badge_type, include_badge_nums=True):
+        self._badge_type = badge_type
+        self._include_badge_nums = include_badge_nums
+
+    def write_row(self, row, out):
+        # plugins can override this hook
+        out.writerow(row)
+
+
+class PersonalizedBadgeReport(ReportBase):
+    """
+    Generate a CSV file which contains personalized badges with custom printed_names on them
+    """
+
+    def run(self, out, session):
+        for a in session.query(sa.Attendee).order_by('badge_num')\
+                .filter(sa.Attendee.badge_num != 0)\
+                .filter(sa.Attendee.badge_type == self._badge_type)\
```

Comment:
Because not all of these reports are actually based on badge type, I'd suggest using a `*filters` param like in https://github.com/magfest/ubersystem/blob/master/uber/models.py#L625. Your function calls will look almost exactly the same, except you can also filter by kick-in level. See https://github.com/magfest/ubersystem/blob/master/uber/site_sections/registration.py#L723 for an example of how to call with a filter.